meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common-alternative.cmd.in

```diff
@@ -98,21 +106,27 @@ if test "${fiovb.is_secondary_boot}" = "0"; then
 				setenv start_blk "${bootloader2_s}"
 				run load_image
 				run update_primary_image2
-				run bootcmd_bootenv
-				if test -n "${fiovb_rpmb}"; then
-					fiovb write_pvalue bootupgrade_available 0;
-					fiovb write_pvalue bootfirmware_version ${bootfirmware_version};
-				else
-					setenv bootupgrade_available 0;
-					setenv bootfirmware_version ${bootfirmware_version};
-				fi
-
-				run saveenv_mmc
+
 				echo "${fio_msg} doing reset to boot newly updated images..."
 				reset
 			fi;
 		fi
 	fi
+	# That means that ak-lite confirmed succesfull boot, so we can now reset
+	# fiovb.bootupgrade_available variable
+	if test "${fiovb.bootupgrade_available}" = "1" && test "${fiovb.upgrade_available}" = "0"; then
+		echo "${fio_msg} finishing boot firmware update..."
+		run bootcmd_bootenv
+		if test -n "${fiovb_rpmb}"; then
+			fiovb write_pvalue bootupgrade_available 0;
+			fiovb write_pvalue bootfirmware_version ${bootfirmware_version};
```

Comment:
Great point.

Reply:
yeah, at first sight it might look like vulnaribility, but value "unfolding" works too simplistic (comparing to real interpreted languages) and can not trigger any code execution. I have even run some tests:

with wrapping:
```
=> setenv bootfirmware_version "0; echo hello"           
=> setenv bootfirmware_version "${bootfirmware_version}";
=> print bootfirmware_version                            
bootfirmware_version=0; echo hello

```
and without:
```
=> setenv bootfirmware_version "0; echo hello"
=> setenv bootfirmware_version ${bootfirmware_version};  
=> print bootfirmware_version                          
bootfirmware_version=0; echo hello

```
in both cases it did not result in execution of `echo` cmd. The only way to do that is use `run` explicitly:
```
=> run bootfirmware_version
Unknown command '0' - try 'help'
hello

```

Reply:
Sure. Now u-boot uses such a simple unfolding process. In the future, it can be replaced with a more advanced one. We can prevent possible side effects right now by just add a pair of quotes :)

Reply:
Yes, makes sense. 
Fixed that